internal/webhooks/iam/v1alpha1/user_webhook.go

```diff
@@ -109,3 +114,28 @@ func (v *UserValidator) createSelfManagePolicyBinding(ctx context.Context, user
 
 	return nil
 }
+
+// createUserPreference creates a UserPreference for the new user
+func (v *UserValidator) createUserPreference(ctx context.Context, user *iamv1alpha1.User) error {
```

Comment:
You'll want to create a new PolicyBinding to grant the user access to get / update this UserPreference resource.

Reply:
Updated @scotwells

Reply:
@mustela you're still missing the creation of the Policy Binding resource after the UserPreference resource is created by the webhook. The policy binding is what would grant the user access to retrieve / update the UserPreference object.

Also just realized you're missing a ProtectedResource definition for the UserPreference CRD so the IAM system will be programmed to support the permissions available. You can see an [example here](https://github.com/datum-cloud/milo/blob/main/config/protected-resources/iam/role.yaml).

Reply:
@scotwells updated!